pkg/master/thirdparty_controller.go

```diff
@@ -37,8 +37,21 @@ func makeThirdPartyPath(group string) string {
 	return thirdpartyprefix + "/" + group
 }
 
-func getThirdPartyGroupName(path string) string {
-	return strings.TrimPrefix(strings.TrimPrefix(path, thirdpartyprefix), "/")
+func makeThirdPartyVersionedPath(group, version string) string {
+	if len(group) == 0 {
+		return thirdpartyprefix
+	}
+	return fmt.Sprintf("%s/%s/%s", thirdpartyprefix, group, version)
+}
+
+func getThirdPartyGroupKind(path string) (string, string) {
+	parts := strings.Split(path, "/")
+	if len(parts) == 4 {
+		return parts[2], parts[3]
+	} else if len(parts) == 5 {
```

Comment:
Can you add a comment with examples here.
For ex: for length 4: apis/group/version/resource.
Am not sure whats an example of length 5?